test/tests/functional/pbs_cray_vnode_pool.py

```diff
@@ -0,0 +1,325 @@
+        since another mom is reporting them.
+        Check that a new inventory mom is listed in the log.
+        Bring up killed mom.
+        """
+        self.server.manager(MGR_CMD_SET, SERVER, {"log_events": -1})
+
+        found_in_momA = 0
+        found_in_momB = 0
+        self.momA = self.moms.values()[0]
+        try:
+            self.momB = self.moms.values()[1]
+        except IndexError:
+            self.assertTrue(
+                False, msg="Provide at least 2 moms while invoking test")
+
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+
+        attr = {'vnode_pool': '1'}
+
+        start_time = int(time.time())
+
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostA, attrib=attr)
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.hostB, attrib=attr)
+
+        rc = self.server.log_match("Mom %s added to vnode_pool %s" %
+                                   (self.hostB, '1'), max_attempts=5,
+                                   starttime=start_time)
+        self.assertTrue(rc)
+
+        rc = self.momA.log_match(
+            "Hello (no inventory required) from server", max_attempts=9,
+            starttime=start_time)
+        if rc is None:
+            found_in_momA = 0
+        else:
+            found_in_momA = 1
+
+        rc = self.momB.log_match(
+            "Hello (no inventory required) from server", max_attempts=9,
+            starttime=start_time)
+        if rc is None:
+            found_in_momB = 0
+        else:
+            found_in_momB = 1
+
+        self.assertEqual(found_in_momA + found_in_momB,
+                         1, msg="an inventory mom not chosen correctly")
+
+        # Only one mom is inventory mom
+        if (found_in_momA == 0):
+            self.inv_mom = self.momA
+            self.noninv_mom = self.momB
+        else:
+            self.inv_mom = self.momB
+            self.noninv_mom = self.momA
+
+        self.logger.info("Inventory mom is %s." % self.inv_mom.shortname)
+        self.logger.info("Non-inventory mom is %s." %
+                         self.noninv_mom.shortname)
+
+        start_time = int(time.time())
+
+        # Kill inventory mom
+        self.inv_mom.stop(sig='-KILL')
+
+        # Check that former inventory mom is down
+        rv = self.server.expect(
+            VNODE, {'state': 'down'}, id=self.inv_mom.shortname,
+            max_attempts=10, interval=2)
+        self.assertTrue(rv)
+
+        # Check if inventory mom changed and is listed in the server log.
+        rc = self.server.log_match(
+            "Setting inventory_mom for vnode_pool %s to %s" %
+            ('1', self.noninv_mom.shortname), max_attempts=5,
+            starttime=start_time)
+        if rc is None:
+            self.logger.info(
+                "Inventory mom %s is not in server logs." %
+                (self.noninv_mom.shortname))
+            self.assertTrue(rc)
```

Comment:
I'd not use the if.  I'd use the msg parameter to self.assertTrue().  You can follow the it with a self.logger of the following message.  The second self.assertTrue() is not needed.